app/src/main/java/com/cmput301f20t21/bookfriends/ui/request/RequestActivity.java

```diff
@@ -0,0 +1,177 @@
+package com.cmput301f20t21.bookfriends.ui.request;
+
+import android.net.Uri;
+import android.os.Bundle;
+import android.view.MenuItem;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.cmput301f20t21.bookfriends.R;
+import com.cmput301f20t21.bookfriends.entities.Book;
+import com.cmput301f20t21.bookfriends.entities.Request;
+import com.cmput301f20t21.bookfriends.ui.library.OwnedListFragment;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RequestActivity extends AppCompatActivity implements ConfirmDialog.ConfirmDialogListener {
+    private RecyclerView recyclerView;
+    private RequestAdapter requestAdapter;
+    private final ArrayList<Request> requestDataList = new ArrayList<>();
```

Comment:
This is a local copy of the requests in the view model? Looks like it will cause confusion when we want to update or remove the request.

Reply:
@kdvuong agreed ^. we should keep all data operation to inside view model as much as we can. What's the reason that we need this local copy of all requests?